airbyte-metrics/reporter/src/main/java/io/airbyte/metrics/reporter/MetricRepository.java

```diff
@@ -0,0 +1,189 @@
+                      SELECT workspace_id, count(c.id) as num_conn
+                      FROM actor
+                        INNER JOIN workspace ws ON actor.workspace_id = ws.id
+                        INNER JOIN connection c ON actor.id = c.source_id
+                      WHERE ws.tombstone = false
+                        AND actor.tombstone = false AND actor.actor_type = 'source'
+                        AND c.status = 'active'
+                      GROUP BY workspace_id;
+                      """;
+    return ctx.fetch(query).getValues("num_conn", long.class);
+  }
+
+  long numScheduledActiveConnectionsInLastDay() {
+    final var queryForTotalConnections = """
+                                         select count(1) as connection_count
+                                         from connection c
+                                         where
+                                           c.updated_at < now() - interval '24 hours 1 minutes'
+                                           and cast(c.schedule::jsonb->'timeUnit' as text) IN ('"hours"', '"minutes"')
+                                           and c.status = 'active'
+                                         """;
+
+    return ctx.fetchOne(queryForTotalConnections).get("connection_count", long.class);
+  }
+
+  long numberOfJobsNotRunningOnScheduleInLastDay() {
+    // This query finds all sync jobs ran in last 24 hours and count how many times they have run.
+    // Comparing this to the expected number of runs (24 hours divide by configured cadence in hours),
+    // if it runs below that expected number it will be considered as abnormal instance.
+    // For example, if it's configured to run every 6 hours but in last 24 hours it only has 3 runs,
+    // it will be considered as 1 abnormal instance.
+    final var queryForAbnormalSyncInHoursInLastDay = """
+                                                     select count(1) as cnt
+                                                     from (
+                                                       select
+                                                         c.id,
+                                                         count(*) as cnt
+                                                       from connection c
+                                                       left join jobs j on j.scope::uuid = c.id
+                                                       where
+                                                         c.schedule is not null
+                                                         and c.schedule != 'null'
+                                                         and j.created_at > now() - interval '24 hours 1 minutes'
+                                                         and c.status = 'active'
+                                                         and j.config_type = 'sync'
+                                                         and c.updated_at < now() - interval '24 hours 1 minutes'
```

Comment:
Why do we need the filter on the updated_at of the connection table?

Reply:
This too was copied verbatim from the [existing implementation](https://github.com/airbytehq/airbyte/pull/17365/files/477451ec7a0a567ca5351edc23caae8b59863e40#diff-5491910fea78d525278084231b60d98ea3242018c44d470e32fefa5df6ffe9f4L206). See my comments above about a follow-up PR.